Dockerfile

```
@@ -1,15 +1,38 @@
+# Install Poetry
+RUN curl -sSL https://install.python-poetry.org | python3 -
+ENV PATH="/root/.local/bin:$PATH"
+
+# Copy Poetry configuration files
+COPY pyproject.toml poetry.lock* ./
+
+# Configure Poetry to not use virtualenvs inside Docker
+RUN poetry config virtualenvs.create false
+
+# Install dependencies without installing the project itself
+RUN poetry install --no-interaction --no-ansi --no-root
```

Hadolint (2.12.0)

[info] Multiple consecutive `RUN` instructions. Consider consolidation.

(DL3059)